Mathlib/Algebra/Algebra/Operations.lean

```diff
@@ -706,6 +706,76 @@ scoped[Pointwise] attribute [instance] Submodule.pointwiseMulSemiringAction
 
 end
 
+section Quotient
+
+/-- The elements of `I / J` are the `x` such that `x • J ⊆ I`.
+
+In fact, we define `x ∈ I / J` to be `∀ y ∈ J, x * y ∈ I` (see `mem_div_iff_forall_mul_mem`),
+which is equivalent to `x • J ⊆ I` (see `mem_div_iff_smul_subset`), but nicer to use in proofs.
+
+This is the general form of the ideal quotient, traditionally written $I : J$.
+-/
+instance : Div (Submodule R A) :=
+  ⟨fun I J =>
+    { carrier := { x | ∀ y ∈ J, x * y ∈ I }
+      zero_mem' := fun y _ => by
+        rw [zero_mul]
+        exact Submodule.zero_mem _
+      add_mem' := fun ha hb y hy => by
+        rw [add_mul]
+        exact Submodule.add_mem _ (ha _ hy) (hb _ hy)
+      smul_mem' := fun r x hx y hy => by
+        rw [Algebra.smul_mul_assoc]
+        exact Submodule.smul_mem _ _ (hx _ hy) }⟩
+
+@[simp]
+theorem mem_div_iff_forall_mul_mem {x : A} {I J : Submodule R A} :
+    x ∈ I / J ↔ ∀ y ∈ J, x * y ∈ I :=
+  Iff.rfl
+
+theorem mem_div_iff_smul_subset {x : A} {I J : Submodule R A} :
+    x ∈ I / J ↔ x • (J : Set A) ⊆ I :=
+  ⟨fun h y ⟨y', hy', xy'_eq_y⟩ => by
+      rw [← xy'_eq_y]
+      exact h _ hy'
+    , fun h _ hy => h (Set.smul_mem_smul_set hy)⟩
+
+@[simp]
+theorem le_div_iff {I J K : Submodule R A} : I ≤ J / K ↔ ∀ x ∈ I, ∀ z ∈ K, x * z ∈ J :=
+  Iff.rfl
+
+theorem le_div_iff_mul_le {I J K : Submodule R A} : I ≤ J / K ↔ I * K ≤ J := by
+  rw [le_div_iff, mul_le]
+
+@[simp, nolint simpNF]
+theorem one_le_one_div {I : Submodule R A} : 1 ≤ 1 / I ↔ I ≤ 1 := by
+  rw [le_div_iff_mul_le, one_mul]
+
+@[simp, nolint simpNF]
+theorem one_mem_div {I J : Submodule R A} : 1 ∈ I / J ↔ J ≤ I := by
+  rw [← one_le, le_div_iff_mul_le, one_mul]
```

Comment:
Note that simp normal form was broken by generalizing these theorems. I added nolints, but I don't know that this is the right solution. I ask reviewers for advice on what to do here.